libslide/src/evaluator_rules/registry.rs

```diff
@@ -0,0 +1,90 @@
+mod fn_rules;
+
+use super::rule::Rule;
+use super::unbuilt_rule::UR;
+
+use fn_rules::*;
+
+// TODO: will allow lint in a future iteration when we support removing rules.
+#[allow(dead_code)]
+pub enum RuleName {
+    Add = 0,
+    Subtract,
+    Multiply,
+    Divide,
+    Modulo,
+    Exponentiate,
+    Posate,
+    Negate,
+    AdditiveIdentity,
+    ReorderConstants,
+    UnwrapParens,
+    UnwrapBraces,
+}
+
+static DEFAULT_RULESET: &[UR] = &[
+    UR::F(add),                           // Add
+    UR::F(subtract),                      // Subtract
+    UR::F(multiply),                      // Multiply
+    UR::F(divide),                        // Divide
+    UR::F(modulo),                        // Modulo
+    UR::F(exponentiate),                  // Exponentiate
+    UR::F(posate),                        // Posate
+    UR::F(negate),                        // Negate
+    UR::S("_a + 0 -> _a"),                // AdditiveIdentity
+    UR::M(&["#a + $b -> $b + #a"]),       // ReorderConstants
+    UR::M(&["($a) -> $a", "(#a) -> #a"]), // UnwrapParens
+    UR::M(&["[$a] -> $a", "[#a] -> #a"]), // UnwrapBraces
+];
+
+/// Set of unbuilt rules.
+pub struct RuleSet {
+    rules: Vec<UR>,
+}
+
+impl Default for RuleSet {
+    /// Constructs the default rule set.
+    fn default() -> Self {
+        Self {
+            rules: DEFAULT_RULESET.to_vec(),
+        }
+    }
+}
+
+impl RuleSet {
+    /// Creates a list of `Rules`s from the unbuilt rule set.
+    pub fn build(&self) -> Vec<Rule> {
+        let num_rules = self.rules.iter().fold(0, |sum, ur| match ur {
+            UR::M(m) => sum + m.len(),
+            _ => sum + 1,
+        });
+
+        let mut built_rules = Vec::with_capacity(num_rules);
+        for unbuilt_rule in self.rules.iter() {
+            match unbuilt_rule {
+                UR::S(s) => built_rules.push(Rule::from_str(s)),
+                UR::M(multiple) => {
+                    for s in multiple.iter() {
+                        built_rules.push(Rule::from_str(s));
+                    }
+                }
+                UR::F(f) => built_rules.push(Rule::from_fn(*f)),
+            }
+        }
+        built_rules
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn builds_rules() {
+        let rule_set = RuleSet::default();
+        let built_rules = rule_set.build();
+        let var_plus_zero = &built_rules[RuleName::AdditiveIdentity as usize];
+
+        assert_eq!(var_plus_zero.to_string(), "_a + 0 -> _a");
+    }
+}
```

Comment:
this should be cleaned up somehow for when we have a large number of rules. I assume you plan on this.

Reply:
Can we also add tseting for multiple rules.

Reply:
Well this function just checks a single rule is built. Would you like to check that we build all rules correctly? It should be easy to do that.

Reply:
yea thats all.

Reply:
Gonna approve, just add a quick test here for building all rules.